app/services/api/beta/search_query_parser_service.rb

```diff
@@ -3,15 +3,16 @@ module Beta
     class SearchQueryParserService
       delegate :client, to: :class
 
-      def initialize(original_search_query)
+      def initialize(original_search_query, spell)
         @original_search_query = original_search_query
+        @spell = spell
       end
 
       def call
         if null_result?
           ::Beta::Search::SearchQueryParserResult::Null.build('original_search_query' => @original_search_query)
         else
-          result_attributes = client.get('tokens', q: @original_search_query).body
+          result_attributes = client.get('tokens', q: @original_search_query, spell:).body
```

Comment:
to be consistent with the param,
`q: @original_search_query, ` 
I'd use 
`client.get('tokens', q: @original_search_query, spell: @spell).body`

(or maybe you want to create another private method for `original_search_query` )

Reply:
I could do that, yeah. :+1: